src/dispatch/scheduler.py

```diff
@@ -7,28 +7,26 @@
 .. moduleauthor:: Kevin Glisson <kglisson@netflix.com>
 .. moduleauthor:: Marc Vilanova <mvilanova@netflix.com>
 """
-import threading
 import logging
 import time
 
+from multiprocessing.pool import ThreadPool
+
 import schedule
 
 log = logging.getLogger(__name__)
 
 
-def run_threaded(job_func):
-    """Runs a given job in a thread."""
-    job_thread = threading.Thread(target=job_func)
-    job_thread.start()
-
-
 #  See: https://schedule.readthedocs.io/en/stable/ for documentation on job syntax
-class Scheduler(object):
+class Scheduler:
     """Simple scheduler class that holds all scheduled functions."""
 
     registered_tasks = []
     running = True
 
+    def __init__(self, num_workers=100):
```

Comment:
Due to my lack of familiarity here... how does this not create a new bottleneck?